packages/cli/src/services/check-runner.ts

```diff
@@ -34,26 +34,37 @@ export type RunLocation = PublicRunLocation | PrivateRunLocation
 
 type CheckRunId = string
 
+type CheckScheduler = (checkRunSuiteId: string) =>
+  Promise<{ testSessionId?: string, testResultIds?: Map<string, string>, checks: Map<CheckRunId, any>}>
+
 export default class CheckRunner extends EventEmitter {
-  project: Project
-  checks: Map<CheckRunId, any>
-  groups: Record<string, CheckGroup>
+  checks?: Map<CheckRunId, any>
   // If there's an error in the backend and no check result is sent, the check run could block indefinitely.
   // To avoid this case, we set a per-check timeout.
   timeouts: Map<CheckRunId, NodeJS.Timeout>
-  location: RunLocation
   accountId: string
-  apiKey: string
   timeout: number
   verbose: boolean
-  shouldRecord: boolean
-  repoInfo?: GitInformation | null
-  environment?: string | null
   queue: PQueue
+  checkScheduler: CheckScheduler
 
-  constructor (
+  private constructor (
+    checkScheduler: CheckScheduler,
+    accountId: string,
+    timeout: number,
+    verbose: boolean,
+  ) {
+    super()
+    this.timeouts = new Map()
+    this.queue = new PQueue({ autoStart: false, concurrency: 1 })
+    this.checkScheduler = checkScheduler
+    this.timeout = timeout
+    this.verbose = verbose
+    this.accountId = accountId
+  }
+
+  static testRunner (
```

Comment:
Why not creating a class that extends CheckRunner? Not sure if like this abstraction

Reply:
Extending `CheckRunner` makes sense, I'll try that.